src/mapping/AxialGeoMultiscaleMapping.cpp

```diff
@@ -116,31 +110,48 @@ void AxialGeoMultiscaleMapping::mapConsistent(const time::Sample &inData, Eigen:
   // We currently don't support 1D data, so we need that the user specifies data of the same dimensions on both sides
   PRECICE_ASSERT(inDataDimensions == outDataDimensions);
 
+  // Effective component (axis) to read/write: 0 for scalar fields (to avoid out-of-bounds), or 0 (x)/ 1 (y)/ 2 (z) for vectors
+  int effectiveCoordinate;
+  if (inDataDimensions == 1) {
+    effectiveCoordinate = 0;
+  } else {
+    effectiveCoordinate = static_cast<std::underlying_type_t<MultiscaleType>>(_axis);
+    PRECICE_ASSERT(effectiveCoordinate == static_cast<std::underlying_type_t<MultiscaleType>>(MultiscaleAxis::X) ||
+                       effectiveCoordinate == static_cast<std::underlying_type_t<MultiscaleType>>(MultiscaleAxis::Y) ||
+                       effectiveCoordinate == static_cast<std::underlying_type_t<MultiscaleType>>(MultiscaleAxis::Z),
+                   "Unknown multiscale axis type.");
+  }
+
   PRECICE_DEBUG("Map consistent");
+
   if (_type == MultiscaleType::SPREAD) {
-    /*
-      3D vertices are assigned a value based on distance from the 1D vertex.
-      Currently, a Hagen-Poiseuille profile determines the velocity value.
-    */
+
     PRECICE_ASSERT(input()->nVertices() == 1);
     size_t const outSize = output()->nVertices();
-
-    for (size_t i = 0; i < outSize; i++) {
-      PRECICE_ASSERT(static_cast<size_t>((i * outDataDimensions) + effectiveCoordinate) < static_cast<size_t>(outputValues.size()), ((i * outDataDimensions) + effectiveCoordinate), outputValues.size());
-      // When adding support for 2D, remember that this should be 1.5 * inputValues(effectiveCoordinate) * (1 - (_vertexDistances[i] * _vertexDistances[i]));
-      outputValues((i * outDataDimensions) + effectiveCoordinate) = 2 * inputValues(effectiveCoordinate) * (1 - (_vertexDistances[i] * _vertexDistances[i]));
+    if (inDataDimensions == 1) {
+      // Scalar: copy same value to all output vertices
+      const double s = inputValues(0);
+      for (size_t i = 0; i < outSize; i++) {
+        outputValues(i) = s;
+      }
+    } else {
+      // Vector: Currently, a Hagen-Poiseuille profile determines the velocity value.
+      for (size_t i = 0; i < outSize; i++) {
+        PRECICE_ASSERT(static_cast<size_t>((i * outDataDimensions) + effectiveCoordinate) < static_cast<size_t>(outputValues.size()),
+                       ((i * outDataDimensions) + effectiveCoordinate), outputValues.size());
+        // When adding support for 2D, remember that this should be 1.5 * inputValues(effectiveCoordinate) * (1 - (_vertexDistances[i] * _vertexDistances[i]));
+        outputValues((i * outDataDimensions) + effectiveCoordinate) = 2 * inputValues(effectiveCoordinate) * (1 - (_vertexDistances[i] * _vertexDistances[i]));
+      }
```

Comment:
We probably should first merge https://github.com/precice/precice/pull/2377 and then reconsider the different profiles here.

I think that distributing a scalar from the 1D to the 3D with a profile should also be a valid option. We can add a "not yet implemented" error, but these should be independent options.

Otherwise, we should check and throw an error earlier if a user tries to set a profile for a scalar spread, and document it.